Sources/BuildSystemIntegration/ExternalBuildSystemAdapter.swift

```diff
@@ -94,10 +94,34 @@ actor ExternalBuildSystemAdapter {
   private var lastRestart: Date?
 
   static package func projectRoot(for workspaceFolder: AbsolutePath, options: SourceKitLSPOptions) -> AbsolutePath? {
-    guard localFileSystem.isFile(workspaceFolder.appending(component: "buildServer.json")) else {
-      return nil
+    var buildServerConfigLocations = [String]()
+
+    #if os(Windows)
+    let workspacePath = workspaceFolder.pathString + "\\.bsp"
```

Comment:
Instead of concatenating strings here, we should use `workspaceFolder.appending(component: ".bsp")` Same below.

Reply:
But `.appending` adds a `/` and for Windows isn't `\` the way to separate directories (although the former works in most cases)?

Reply:
`appending(component:)` should take care of all platform differences in path representations.